### ISSUE TITLE
New constant score parse error

### ISSUE THREAD

**Issue by maedoc** [elastic/elasticsearch-dsl-py#649]
We're deploying from PyPI and saw fairly quickly an issue with `Q('constant_score', ...)`, which appears to be introduced by commit 88da29338c566dcc322fe9438c96c2932937d0e7, to wit.

In Kibana Dev Console, 
```json
GET /

{
...
  "version": {
    "number": "5.2.2",
    "build_hash": "f9d9b74",
    "build_date": "2017-02-24T17:26:45.835Z",
    "build_snapshot": false,
    "lucene_version": "6.4.1"
  },
  "tagline": "You Know, for Search"
}
```

and an array valued `filter` (generated/introduced by the commit in question), 
```
GET _search
{
  "query": {
    "constant_score": {
      "filter": [{}],
    }
  }
}

{
  "error": {
    "root_cause": [
      {
        "type": "parsing_exception",
        "reason": "unexpected token [START_ARRAY]",
        "line": 4,
        "col": 17
      }
    ],
    "type": "parsing_exception",
    "reason": "unexpected token [START_ARRAY]",
    "line": 4,
    "col": 17
  },
  "status": 400
}
```

Our workaround is to have a compatability `ConstantScore` class in our codebase.

I see the unit tests are only testing that a dictionary can be produced, but not that ES will actually consume the result, which is why the tests pass.

**Comment by tylerjharden** [comment]
I fixed this here: https://github.com/elastic/elasticsearch-dsl-py/pull/657

Awaiting the maintainers to approve and merge.

**Comment by HonzaKral** [comment]
closed via #657, thanks!

**Comment by maedoc** [comment]
thanks @tylerjharden
